pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -1339,6 +1396,118 @@ func (c *controller) UpdateNodeTerminationCondition(ctx context.Context, machine
 	return err
 }
 
+func (c *controller) updateMachineToFailedState(ctx context.Context, description string, machine, clone *v1alpha1.Machine) (machineutils.RetryPeriod, bool, error) {
+	// Log the error message for machine failure
+	klog.Error(description)
+
+	clone.Status.LastOperation = v1alpha1.LastOperation{
+		Description:    description,
+		State:          v1alpha1.MachineStateFailed,
+		Type:           machine.Status.LastOperation.Type,
+		LastUpdateTime: metav1.Now(),
+	}
+	clone.Status.CurrentStatus = v1alpha1.CurrentStatus{
+		Phase: v1alpha1.MachineFailed,
+		//TimeoutActive:  false,
+		LastUpdateTime: metav1.Now(),
+	}
+
+	_, err := c.controlMachineClient.Machines(clone.Namespace).UpdateStatus(ctx, clone, metav1.UpdateOptions{})
+	updated := false
+	if err != nil {
+		// Keep retrying until update goes through
+		klog.Errorf("update failed for machine %q in function. Retrying, error: %q", machine.Name, err)
+	} else {
+		updated = true
+		klog.Infof("Machine State has been updated for %q with providerID %q and backing node %q", machine.Name, getProviderID(machine), getNodeName(machine))
+		// Return error for continuing in next iteration
+		err = fmt.Errorf("machine creation is successful. Machine State has been UPDATED")
+	}
+
+	return machineutils.ShortRetry, updated, err
```

Comment:
If you are always going to return `machineutils.ShortRetry` then why bother returning this from this method. The caller should do that instead. It would make sense to return it in case retry could have different values or no value (in a specific condition). Any reason to always return `machineutils.ShortRetry`?